core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -531,6 +528,9 @@ public void delete(Engine.Delete delete) {
             throw ex;
         }
         indexingService.postDelete(delete);
```

Comment:
> can remove the dependency between IndexShard and IndexMemoryController 

I like this in principle but does this mean I should have IMC call `.addListener` for each shard so it's invoked on each indexing op?  This seems tricky (I somehow need to do this as soon as the shard is created before any indexing ops are sent to it).  Or can I just call `IMC.bytesWritten` directly inside `ShardIndexingService.postIndex,Delete`?